satpy/readers/lma_l2.py

```diff
@@ -0,0 +1,158 @@
+            if matches and var_name in self:
+                logger.debug("Handling previously configured variable: %s", var_name)
+                handled_variables.add(var_name)
+                new_info = ds_info.copy()  # don't mess up the above yielded
+                yield True, new_info
+            elif is_avail is None:
+                # if we didn't know how to handle this dataset and no one else did
+                # then we should keep it going down the chain
+                yield is_avail, ds_info
+
+        # Iterate over dataset contents
+        for var_name, val in self.file_content.items():
+            # Only evaluate variables
+            if isinstance(val, netCDF4.Variable):
+                logger.debug("Evaluating new variable: %s", var_name)
+                var_shape = self[var_name + "/shape"]
+                logger.debug("Dims:{}".format(var_shape))
+                if var_shape == (lat_shape, lon_shape):
+                    logger.debug("Found valid additional dataset: %s", var_name)
+                    # Skip anything we have already configured
+                    if var_name in handled_variables:
+                        logger.debug("Already handled, skipping: %s", var_name)
+                        continue
+                    handled_variables.add(var_name)
+                    # Create new ds_info object
+                    new_info = {
+                        'name': var_name,
+                        'file_key': var_name,
+                        'file_type': self.filetype_info['file_type'],
+                    }
+                    logger.debug(var_name)
+                    yield True, new_info
+
+    def get_dataset(self, ds_id, info):
+        """Load dataset designated by the given key from file."""
+        logger.debug("Getting data for: %s", ds_id['name'])
+        file_key = info.get('file_key', ds_id['name'])
+        data = np.flipud(self[file_key])
+        data = xr.DataArray(data, dims=[1, 'y', 'x'])
+        data.attrs = self.get_metadata(data, info)
+
+        if 'lon' in data.dims:
+            data.rename({'lon': 'x'})
+        if 'lat' in data.dims:
+            data.rename({'lat': 'y'})
+        return data
+
+    def get_area_def(self, dsid):
+        """Flip data up/down and define equirectangular AreaDefintion."""
+        flip_lat = np.flipud(self['latArr'])
+        latlon = np.meshgrid(self['lonArr'], flip_lat)
+
+        width = self['lonArr/shape'][0]
+        height = self['latArr/shape'][0]
+
+        lower_left_x = latlon[0][height-1][0]
+        lower_left_y = latlon[1][height-1][0]
+
+        upper_right_y = latlon[1][0][width-1]
+        upper_right_x = latlon[0][0][width-1]
+
+        area_extent = (lower_left_x, lower_left_y, upper_right_x, upper_right_y)
+        description = "Trail data Projection"
+        area_id = 'lma'
+        proj_id = 'equirectangular'
+        proj_dict = {'proj': 'longlat', 'datum': 'WGS84', 'ellps': 'WGS84', }
```

Comment:
From what I saw in the ncdump from the file you had, the flattening ratio is provided in the data, so maybe you want to use that instead of WGS84?